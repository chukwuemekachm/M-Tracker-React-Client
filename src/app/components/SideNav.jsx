import React from 'react';
import PropTypes from 'prop-types';

const SideNav = ({ requests, handleNavigation }) => (
  <ul className="list-group mb-3">
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
      data-toggle="modal"
      data-target="#createRequest"
    >
      Create Request
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
      onClick={() => handleNavigation('all')}
    >
      All Requests
      <span className="badge badge-primary badge-pill">
        {requests ? requests.length : 0}
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
      onClick={() => handleNavigation('pending')}
    >
      Pending Requests
      <span className="badge badge-primary badge-pill">
        {requests ? requests.filter(request => request.status === 'pending').length : 0}
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
      onClick={() => handleNavigation('approved')}
    >
      Approved Requests
      <span className="badge badge-primary badge-pill">
        {requests ? requests.filter(request => request.status === 'approved').length : 0}
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
      onClick={() => handleNavigation('resolved')}
    >
      Resolved Requests
      <span className="badge badge-primary badge-pill">
        {requests ? requests.filter(request => request.status === 'resolved').length : 0}
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
      onClick={() => handleNavigation('disapproved')}
    >
      Dis-approved Requests
      <span className="badge badge-primary badge-pill">
        {requests ? requests.filter(request => request.status === 'disapproved').length : 0}
      </span>
    </li>
  </ul>
);

SideNav.propTypes = {
  requests: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  handleNavigation: PropTypes.func.isRequired,
};

export default SideNav;
