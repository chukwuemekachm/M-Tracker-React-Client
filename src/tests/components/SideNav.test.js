import React from 'react';
import { shallow } from 'enzyme';
import SideNav from '../../app/components/SideNav';

const requests = [
  { title: 'first request', status: 'approved', createdat: '2018-09-04T18:11:56.976Z' },
  { title: 'first request', status: 'disapproved', createdat: '2018-09-04T18:11:56.976Z' },
  { title: 'first request', status: 'resolved', createdat: '2018-09-04T18:11:56.976Z' },
  { title: 'first request', status: 'pending', createdat: '2018-09-04T18:11:56.976Z' },
];
const mockFunction = jest.fn();

describe('Side nav component', () => {
  it('should render without errors when requests are present', () => {
    const wrapper = shallow(<SideNav requests={requests} handleNavigation={mockFunction} />);
    expect(wrapper.find('ul')).toHaveLength(1);
    expect(wrapper.find('li')).toHaveLength(6);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without errors when no request is present', () => {
    const wrapper = shallow(<SideNav requests={[]} handleNavigation={mockFunction} />);
    expect(wrapper.find('ul')).toHaveLength(1);
    expect(wrapper.find('li')).toHaveLength(6);
  });

  it('should handle all nav button click', () => {
    const wrapper = shallow(<SideNav requests={[]} handleNavigation={mockFunction} />);
    wrapper.find('[name="all"]').simulate('click');
  });

  it('should handle pending nav button click', () => {
    const wrapper = shallow(<SideNav requests={[]} handleNavigation={mockFunction} />);
    wrapper.find('[name="pending"]').simulate('click');
  });

  it('should handle resolved nav button click', () => {
    const wrapper = shallow(<SideNav requests={[]} handleNavigation={mockFunction} />);
    wrapper.find('[name="resolved"]').simulate('click');
  });

  it('should handle approved nav button click', () => {
    const wrapper = shallow(<SideNav requests={[]} handleNavigation={mockFunction} />);
    wrapper.find('[name="approved"]').simulate('click');
  });

  it('should handle disapproved nav button click', () => {
    const wrapper = shallow(<SideNav requests={[]} handleNavigation={mockFunction} />);
    wrapper.find('[name="disapproved"]').simulate('click');
  });
});
