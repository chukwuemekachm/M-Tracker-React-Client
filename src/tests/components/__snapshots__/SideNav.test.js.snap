// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side nav component should render without errors 1`] = `
<ul
  className="list-group mb-3"
>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
    data-target="#createRequest"
    data-toggle="modal"
  >
    Create Request
  </li>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
    onClick={[Function]}
  >
    All Requests
    <span
      className="badge badge-primary badge-pill"
    >
      4
    </span>
  </li>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
    onClick={[Function]}
  >
    Pending Requests
    <span
      className="badge badge-primary badge-pill"
    >
      1
    </span>
  </li>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
    onClick={[Function]}
  >
    Approved Requests
    <span
      className="badge badge-primary badge-pill"
    >
      1
    </span>
  </li>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
    onClick={[Function]}
  >
    Resolved Requests
    <span
      className="badge badge-primary badge-pill"
    >
      1
    </span>
  </li>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
    onClick={[Function]}
  >
    Dis-approved Requests
    <span
      className="badge badge-primary badge-pill"
    >
      1
    </span>
  </li>
</ul>
`;
